scipy/signal/filter_design.py

```diff
@@ -436,7 +438,8 @@ def freqz(b, a=1, worN=512, whole=False, plot=None, fs=2*pi):
         if N < 0:
             raise ValueError('worN must be nonnegative, got %s' % (N,))
         lastpoint = 2 * pi if whole else pi
-        w = np.linspace(0, lastpoint, N, endpoint=False)
+        # if nyquist is true, w should include end point
+        w = np.linspace(0, lastpoint, N, endpoint=nyquist)
```

Comment:
As per the idea above:
```suggestion
        w = np.linspace(0, lastpoint, N, endpoint=nyquist and not whole)
```

Reply:
@larsoner  I fixed the code based on your reviews.